### PR TITLE
Add support for linux-aarch64 w/ cross-compilation

### PR DESCRIPTION
**Pull request by nehaljwani**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mr-c** [comment]
@nehaljwani Cool, just saw this PR. I also could not get linux+arm64+py3.10 to work, so I had that fallback to a pure source build (better than nothing). Perhaps something to report upstream?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ build:
   script:
     - export MYPY_USE_MYPYC=1  # [not (win or (osx and arm64))]
     - set MYPY_USE_MYPYC=1     # [win]
+    - export MYPYC_OPT_LEVEL=2  # [aarch64 and linux]
```
Comment (mr-c) [inline recipe/meta.yaml:21]: If this ends up working for py3.10, can that be restricted to only ` [aarch64 and linux and py=310]`?

Comment (nehaljwani) [reply recipe/meta.yaml:21]: I'm just experimenting at the moment. Will make restrictions later.

Comment (mr-c) [reply recipe/meta.yaml:21]: :+1:, just checking

I'm the maintainer for Debian's package of `mypy`: we also had to play around with the settings to get it to self-compile everywhere

We also run more build-time tests that this recipe currently does; feel free to steal from our scripts

https://salsa.debian.org/med-team/mypy/-/blob/master/debian/rules#L17
https://salsa.debian.org/med-team/mypy/-/blob/master/debian/changelog
https://salsa.debian.org/med-team/mypy/-/blob/master/debian/tests/run-unit-test ("in situ" tests that are run via ci.debian.net on installed packages)

Comment (mr-c) [reply recipe/meta.yaml:21]: https://salsa.debian.org/med-team/mypy/-/blob/master/debian/patches/verbose may also be of interest

Comment (nehaljwani) [reply recipe/meta.yaml]: Reported the issue upstream: https://github.com/python/mypy/issues/11507

Comment (nehaljwani) [reply recipe/meta.yaml]: Thank you for sharing the links, I'll open an issue to add more tests in the recipe.

**Pull request merged**